nms/app/components/insights/AsyncMetric.tsx

```diff
@@ -204,34 +204,27 @@ function getUnit(timeRange: TimeRange) {
   return RANGE_VALUES[timeRange].unit;
 }
 
-function getStepUnit(
-  startEnd: [moment.Moment, moment.Moment],
-): [string, TimeUnit] {
+function getStepUnit(startEnd: [Date, Date]): [string, TimeUnit] {
   const [start, end] = startEnd;
-  const d = moment.duration(end.diff(start));
-  const hrs = d.asHours();
-  const days = d.asDays();
-  let r: RangeValue;
-  if (hrs <= 24) {
-    if (hrs <= 3) {
-      r = RANGE_VALUES['3_hours'];
-    } else if (hrs <= 6) {
-      r = RANGE_VALUES['6_hours'];
-    } else if (hrs <= 12) {
-      r = RANGE_VALUES['12_hours'];
-    } else {
-      r = RANGE_VALUES['24_hours'];
-    }
-  } else {
-    if (days <= 7) {
-      r = RANGE_VALUES['7_days'];
-    } else if (days <= 14) {
-      r = RANGE_VALUES['14_days'];
-    } else {
-      r = RANGE_VALUES['30_days'];
-    }
+  const durationInHours = differenceInHours(end, start);
+  const durationInDays = differenceInDays(end, start);
+
+  let range: RangeValue = RANGE_VALUES['3_hours'];
+  if (durationInHours > 3) {
```

Comment:
I think something went wrong with the change here. If the duration is 10 hours, we we would still go into the first branch now?

Reply:
Okay, this version would work but seem a little confusing...

Actually, in the current version, is the inversion still necessary? I think we could go back to the old version since durationInHours will never be null...

Reply:
Those comments are probably to some intermediate states I had during development.
The basic underlying problem is, that objects have no guaranteed order, such that a `next` item cannot be safely determined. Reconsidering that, I think I came up with a new and more satisfying solution. 
:arrow_right: I re-requested review now.